package.json

```diff
@@ -8,7 +8,9 @@
     "install": "npm install --prefix packages/openactive-broker-microservice && npm install --prefix packages/openactive-integration-tests",
     "prettier": "prettier --check \"packages/**/*.js\" \"packages/**/*.json\"",
     "prettier:write": "prettier --write \"packages/**/*.js\" \"packages/**/*.json\"",
-    "start": "npm start --prefix packages/openactive-broker-microservice",
+    "start": "echo 'Error: please select a package to run and then run it e.g. `npm start --prefix packages/openactive-broker-microservice`' && exit 1",
```

Comment:
`npm start` in the root could run `run-specific-test.sh`? Is that what you meant by collating?

Reply:
@nickevansuk yeah that makes sense. I hadn't completed my PR description, sorry.

I wanted to have an `npm start` that could be run by developer or CI alike, which would just do the right thing based on the environment (e.g. config files). But that doesn't work for CI because there's scenarios like (1). Run with randomized (2). Run with controlled, which would require a way of specifying, in the environment, multiple runs - perhaps each with distinct configuration - which run with the microservice started before and stopped after all runs have completed.

Therefore, that's not possible right now without a design change. I like your suggestion though. It's at least a good scenario for a developer. Maybe just running the post-install section of `simple-ci.sh`, which just runs all the tests

```sh
# Start broker microservice in the background
npm start --prefix packages/openactive-broker-microservice &
pid=$!

# Kill broker microservice in case of error
trap 'err=$?; echo >&2 "Exiting on error $err"; kill $pid; exit $err' ERR

# Run tests
npm start --prefix packages/openactive-integration-tests

# Kill broker microservice on success
kill $pid
```

Reply:
if we add `"$@"` at the end of the line `npm start --prefix packages/openactive-integration-tests` to add the flexibility of passing through the arguments, that would work right?

Reply:
And in fact we could then run it from CI also, if NODE_CONFIG prefix gets passed through.

For scenarios like (1). Run with randomized (2). Run with controlled, it's fine to stop and restart the microservice each time

Reply:
Maybe we do this as above for `npm start` but leave the GitHub Actions CI as-is for now? Only because separating out the microservice from the integration tests is useful to debug where the issues are coming from - and it seems to filter the output from each rather than combining the output together (which makes the console harder to read)

Reply:
![tenor](https://user-images.githubusercontent.com/2067438/87815474-92e1b280-c85d-11ea-8e07-e2f9998dc157.gif)

Reply:
@nickevansuk I'm going to do that on a separate branch (https://github.com/openactive/openactive-test-suite/pull/107, incomplete)